test/OpenTelemetry.Instrumentation.EventCounters.Tests/EventCountersMetricsTests.cs

```diff
@@ -243,7 +209,9 @@ public void InstrumentNameTooLong()
 
         // Act
         connections.Increment(1);
-        AwaitExport(meterProvider, metricItems);
+
+        Task.Delay(1800).Wait();
```

Comment:
Why does this test have a Task.Delay, but the others have the SpinWait?


Also, not your code, but I don't understand what this test is doing.
name is "InstrumentNameTooLong", and it creates an eventName with 100 "e" characters.
But the asserts don't inspect that name nor that name's length.

I think this test needs to be rewritten.
If the Task.Delay fails and no metrics are exported, this test will pass because the foreach on an empty collection.

Reply:
@mic-max - Could you open an issue for this? And disable this test for now.

Reply:
These tests are checking that nothing gets exported for the event counters created in it. But I do agree it is a little weird especially now that others will use `SpinWait`.

It's checking the last part "it will be validated and ignored later in the pipeline" of `GetInstrumentName`

```csharp
    /// <summary>
    /// If the resulting instrument name is too long, it trims the event source name
    /// to fit in as many characters as possible keeping the event name intact.
    /// E.g. instrument for `Microsoft-AspNetCore-Server-Kestrel`, `tls-handshakes-per-second`
    /// would be too long (64 chars), so it's shortened to `ec.Microsoft-AspNetCore-Server-Kestre.tls-handshakes-per-second`.
    ///
    /// If there is no room for event source name, returns `ec.{event name}` and
    /// if it's still too long, it will be validated and ignored later in the pipeline.
    /// </summary>
    private static string GetInstrumentName(string sourceName, string eventName)
```

Reply:
Skipping unit test for now: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/936